pytest_django/plugin.py

```diff
@@ -120,22 +113,30 @@ def _handle_import_error(extra_message):
 
 
 def _add_django_project_to_path(args):
-    args = [x for x in args if not str(x).startswith("-")]
-
-    if not args:
-        args = [py.path.local()]
-
-    for arg in args:
-        arg = py.path.local(arg)
-
-        for base in arg.parts(reverse=True):
-            manage_py_try = base.join('manage.py')
-
-            if _exists(manage_py_try):
-                sys.path.insert(0, str(base))
-                return PROJECT_FOUND % base
+    def is_django_project(path):
+        return path.is_dir() and (path / 'manage.py').exists()
+
+    def find_django_path(args):
+        args = [pathlib.Path(x) for x in args if not str(x).startswith("-")]
+        args = [p for p in args if p.is_dir()]
+
+        if not args:
+            args = [pathlib.Path.cwd()]
+
+        for arg in args:
+            if is_django_project(arg):
+                return arg
+            for parent in arg.parents:
+                if is_django_project(parent):
+                    return parent
```

Comment:
@voidus 
This (looking at parent) is a behavior change already, isn't it?
(at least it is not covered with tests)

Reply:
Oh, sorry.. `arg.parts(reverse=True)` handled this before